includes/class-admin-settings.php

```diff
@@ -0,0 +1,516 @@
+<?php
+/**
+ * The Class for Admin Settings Page and functions to access Settings Values.
+ *
+ * @package aspire-update
+ */
+
+namespace AspirePress;
+
+/**
+ * The Class for Admin Settings Page and functions to access Settings Values.
+ */
+class Admin_Settings {
+
+	/**
+	 * Hold a single instance of the class.
+	 *
+	 * @var object
+	 */
+	private static $instance = null;
+
+	/**
+	 * The Name of the Option Group.
+	 *
+	 * @var string
+	 */
+	private $option_group = 'aspirepress_settings';
+
+	/**
+	 * The Name of the Option.
+	 *
+	 * @var string
+	 */
+	private $option_name = 'aspirepress_settings';
+
+	/**
+	 * An Array containing the values of the Options.
+	 *
+	 * @var array
+	 */
+	private $options = null;
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+		add_action( 'admin_menu', array( $this, 'register_admin_menu' ) );
+		add_action( 'admin_init', array( $this, 'reset_settings' ) );
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'admin_notices', array( $this, 'reset_admin_notice' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ) );
+	}
+
+	/**
+	 * Initialize Class.
+	 *
+	 * @return object
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Get the default values for the settings.
+	 *
+	 * @return array The default values.
+	 */
+	private function get_default_settings() {
+		$options              = array();
+		$options['api_hosts'] = array( 'api.aspirecloud.org' );
+		return $options;
+	}
+
+	/**
+	 * Handles the Reset Functionality and triggers a Notice to inform the same.
+	 *
+	 * @return void
+	 */
+	public function reset_settings() {
+		if (
+			isset( $_GET['reset'] ) &&
+			( 'reset' === $_GET['reset'] ) &&
+			isset( $_GET['reset-nonce'] ) &&
+			wp_verify_nonce( sanitize_key( $_GET['reset-nonce'] ), 'aspirepress-reset-nonce' )
+		) {
+			$options = $this->get_default_settings();
+			update_option( $this->option_name, $options );
+			update_option( 'aspirepress-reset', 'true' );
+
+			wp_safe_redirect(
+				add_query_arg(
+					array(
+						'reset-success'       => 'success',
+						'reset-success-nonce' => wp_create_nonce( 'aspirepress-reset-success-nonce' ),
+					),
+					admin_url( 'options-general.php?page=aspirepress-settings' )
+				)
+			);
+			exit;
+		}
+	}
+
+	/**
+	 * The Admin Notice to convey a Reset Operation has happened.
+	 *
+	 * @return void
+	 */
+	public function reset_admin_notice() {
+		if (
+			( 'true' === get_option( 'aspirepress-reset' ) ) &&
+			isset( $_GET['reset-success'] ) &&
+			( 'success' === $_GET['reset-success'] ) &&
+			isset( $_GET['reset-success-nonce'] ) &&
+			wp_verify_nonce( sanitize_key( $_GET['reset-success-nonce'] ), 'aspirepress-reset-success-nonce' )
```

Comment:
wp_unslash() here too?

Reply:
We are not using any parameters there, just making sure its originating from our site.